server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/UserInfoUtils.java

```diff
@@ -31,4 +31,17 @@ public static String getUsername(String userInfo) {
         }
     }
 
+    /***
+     * Checks if a field "id" exists in the object created from the Json. This will be used to validate the incoming json in POST API calls.
+     * If present, it should reject the API call because new ids are created by MongoDB, and user shouldn't update the document through a
+     * POST call.
+     * @param id
+     */
+    public static void throwExceptionIfIdPresent(String id){
+        if(id != null){
+            throw new IncoreHTTPException(Response.Status.BAD_REQUEST, "Ids are auto-generated by the system. " +
+                "Setting an id is not allowed");
```

Comment:
maybe the message could be a little bit more explicit, to let user know they included "id" field in their POST body and we forbid them from doing that.